python/lsst/daf/persistence/butler.py

```diff
@@ -229,35 +437,49 @@ def getKeys(self, datasetType=None, level=None):
         @returns (dict) valid data id keys; values are corresponding types.
         """
         datasetType = self._resolveDatasetTypeAlias(datasetType)
-        return self.repository.getKeys(datasetType, level)
+
+        keys = None
+        for repoData in self._repos.inputs():
+            if not tag or len(tag.intersection(repoData.tags)) > 0:
```

Comment:
Are we certain `tag` is a set here?  This looks like a public method, and docs seem to suggest tags can be anything.